### PR TITLE
feat: select decimals as strings if cds.env.features.string_decimals

### PR DESCRIPTION
**Pull request by sjvans**

### CODE REVIEW

File: hana/lib/HANAService.js [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ class HANAService extends SQLService {
       // Reading int64 as string to not loose precision
       Int64: expr => `TO_NVARCHAR(${expr})`,
       // Reading decimal as string to not loose precision
-      Decimal: expr => `TO_NVARCHAR(${expr})`,
+      Decimal: cds.env.features.string_decimals ? expr => `TO_NVARCHAR(${expr})` : undefined,
```
Comment (sjvans) [inline hana/lib/HANAService.js:1012]:
```suggestion
      // REVISIT: always cast to string in next major
      Decimal: cds.env.features.string_decimals ? expr => `TO_NVARCHAR(${expr})` : undefined,
```

File: hana/lib/HANAService.js [hunk 2]
```diff
@@ -1009,7 +1009,7 @@ class HANAService extends SQLService {
       // Reading int64 as string to not loose precision
       Int64: expr => `TO_NVARCHAR(${expr})`,
       // Reading decimal as string to not loose precision
```
Comment (sjvans) [inline hana/lib/HANAService.js:1011]:
```suggestion
      // REVISIT: always cast to string in next major
      // Reading decimal as string to not loose precision
```

File: postgres/lib/PostgresService.js [hunk 3]
```diff
@@ -490,7 +490,7 @@ GROUP BY k
       // Reading int64 as string to not loose precision
       Int64: expr => `cast(${expr} as varchar)`,
       // Reading decimal as string to not loose precision
```
Comment (sjvans) [inline postgres/lib/PostgresService.js:492]:
```suggestion
      // REVISIT: always cast to string in next major
      // Reading decimal as string to not loose precision
```

File: sqlite/lib/SQLiteService.js [hunk 4]
```diff
@@ -226,7 +226,7 @@ class SQLiteService extends SQLService {
       Int64: expr => `CAST(${expr} as TEXT)`,
       
       // Reading decimal as string to not loose precision
```
Comment (sjvans) [inline sqlite/lib/SQLiteService.js:228]:
```suggestion
      // REVISIT: always cast to string in next major
      // Reading decimal as string to not loose precision
```

**Comment by BobdenOs** [comment]
Looks good. The tests are just failing as the flag should already be enabled.

File: test/compliance/resources/db/basic/literals/basic.literals.number.js [hunk 5]
```diff
@@ -16,31 +16,27 @@ module.exports = [
   },
   {
     decimal: '3.14153',
-    '=decimal': /^3.1415/
+    '=decimal': /3\.1415/
```
Comment (sjvans) [inline test/compliance/resources/db/basic/literals/basic.literals.number.js:19]:
```suggestion
    '=decimal': /^3\.1415/
```

File: test/compliance/resources/db/basic/literals/basic.literals.number.js [hunk 6]
```diff
@@ -16,31 +16,27 @@ module.exports = [
...
   },
-  /*{
-    decimal: '12345678910', // TODO: should properly consider precision
-    '!': 'should throw an error'
-  },*/
   {
     decimal: 3.14,
-    '=decimal': /^3.14/
+    '=decimal': /3\.14/
```
Comment (sjvans) [inline test/compliance/resources/db/basic/literals/basic.literals.number.js:23]:
```suggestion
    '=decimal': /^3\.14/
```

File: test/compliance/resources/db/basic/literals/basic.literals.number.js (hunk 5 above)
Comment (BobdenOs) [inline test/compliance/resources/db/basic/literals/basic.literals.number.js:19]: I would keep the `^` as the value is only allowed to be longer in the fractionals. The `\` is a good catch.

Comment (sjvans) [inline test/compliance/resources/db/basic/literals/basic.literals.number.js:19]:
```suggestion
    '=decimal': /^3\.1415/
```

File: test/compliance/resources/db/basic/literals/basic.literals.number.js (hunk 6 above)
Comment (sjvans) [inline test/compliance/resources/db/basic/literals/basic.literals.number.js:23]:
```suggestion
    '=decimal': /^3\.14/
```

File: test/compliance/resources/db/basic/literals/basic.literals.number.js [hunk 7]
```diff
@@ -16,31 +16,27 @@ module.exports = [
...
   },
   {
     double: 3.14159265358979
   },
   {
     float: '3.14159265358979',
-    '=float': /^3\.14159265358979/
+    '=float': /3\.14159265358979/
```
Comment (sjvans) [inline test/compliance/resources/db/basic/literals/basic.literals.number.js:30]:
```suggestion
    '=float': /^3\.14159265358979/
```

**Review by BobdenOs — APPROVED** [review]

**Comment by sjvans** [comment]
yes, new adapter comes with string decimals

**Pull request merged**